### PR TITLE
Ikasan 2075 scheduler agent log file management

### PR DESCRIPTION
**Pull request by SerkanGuler**

### CODE REVIEW

File: ikasaneip/ootb/module/scheduler-agent/jar/pom.xml [hunk 1]
```diff
@@ -15,6 +15,12 @@
     <description>Describe the purpose of this Integration Module</description>
 
     <dependencies>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-compress</artifactId>
+            <version>1.20</version>
```
Comment (mick-stewart73) [inline ikasaneip/ootb/module/scheduler-agent/jar/pom.xml]: Please move the version into the parent pom as per general pattern.

File: ikasaneip/ootb/module/scheduler-agent/jar/src/main/java/org/ikasan/ootb/scheduler/agent/module/SchedulerAgentModuleFactory.java [hunk 2]
```diff
@@ -83,6 +83,7 @@ public Module createModule()
                 .withType(ModuleType.SCHEDULER_AGENT)
                 .withFlowFactory(schedulerAgentFlowFactory)
                 .setConfiguration(configuration)
+                .addFlow(schedulerAgentFlowFactory.create("", "HOUSEKEEP_LOG").get(0))
```
Comment (mick-stewart73) [inline ikasaneip/ootb/module/scheduler-agent/jar/src/main/java/org/ikasan/ootb/scheduler/agent/module/SchedulerAgentModuleFactory.java]: This should not be here as this is a configured module.

File: ikasaneip/ootb/module/scheduler-agent/jar/src/main/java/org/ikasan/ootb/scheduler/agent/module/boot/components/HousekeepLogFilesFlowComponentFactory.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    String moduleName;
+
+    @Resource
+    BuilderFactory builderFactory;
+
+
+    public Consumer getScheduledConsumer()
+    {
+        ScheduledConsumerConfiguration configuration = new ScheduledConsumerConfiguration();
+        configuration.setCronExpression("20 20 03 * * ?");
```
Comment (mick-stewart73) [inline ikasaneip/ootb/module/scheduler-agent/jar/src/main/java/org/ikasan/ootb/scheduler/agent/module/boot/components/HousekeepLogFilesFlowComponentFactory.java]: please make the cron configurable from the properties file

File: ikasaneip/ootb/module/scheduler-agent/jar/src/test/resources/application.properties [hunk 4]
```diff
@@ -83,3 +83,10 @@ ikasan.exceptions.retry-configs.[0].delayInMillis=5000
 ikasan.exceptions.retry-configs.[0].maxRetries=-1
 
 ikasan.exceptions.excludedClasses[0]=org.ikasan.spec.component.transformation.TransformationException
+
+# Housekeep Log Files Flow
+housekeep.log.files.process.log-folder=./logs
+housekeep.log.files.process.ttl=25
```
Comment (mick-stewart73) [inline ikasaneip/ootb/module/scheduler-agent/jar/src/test/resources/application.properties]: please change the name of the config value to describe the number value. Is it days for instance?
housekeep.log.files.process.ttl.days

File: ikasaneip/ootb/module/scheduler-agent/jar/pom.xml (hunk 1 above)
Comment (SerkanGuler) [reply ikasaneip/ootb/module/scheduler-agent/jar/pom.xml]: done

File: ikasaneip/ootb/module/scheduler-agent/jar/src/main/java/org/ikasan/ootb/scheduler/agent/module/SchedulerAgentModuleFactory.java (hunk 2 above)
Comment (SerkanGuler) [reply ikasaneip/ootb/module/scheduler-agent/jar/src/main/java/org/ikasan/ootb/scheduler/agent/module/SchedulerAgentModuleFactory.java]: removed

File: ikasaneip/ootb/module/scheduler-agent/jar/src/main/java/org/ikasan/ootb/scheduler/agent/module/boot/components/HousekeepLogFilesFlowComponentFactory.java (hunk 3 above)
Comment (SerkanGuler) [reply ikasaneip/ootb/module/scheduler-agent/jar/src/main/java/org/ikasan/ootb/scheduler/agent/module/boot/components/HousekeepLogFilesFlowComponentFactory.java]: done

File: ikasaneip/ootb/module/scheduler-agent/jar/src/test/resources/application.properties (hunk 4 above)
Comment (SerkanGuler) [reply ikasaneip/ootb/module/scheduler-agent/jar/src/test/resources/application.properties]: done

File: ikasaneip/ootb/module/scheduler-agent/jar/src/main/java/org/ikasan/ootb/scheduler/agent/module/configuration/HousekeepLogFilesProcessConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
+package org.ikasan.ootb.scheduler.agent.module.configuration;
```
Comment (mick-stewart73) [inline ikasaneip/ootb/module/scheduler-agent/jar/src/main/java/org/ikasan/ootb/scheduler/agent/module/configuration/HousekeepLogFilesProcessConfiguration.java]: Sorry one more Serkan. This configuration should live under components/endpoint/configuration

Comment (SerkanGuler) [reply ikasaneip/ootb/module/scheduler-agent/jar/src/main/java/org/ikasan/ootb/scheduler/agent/module/configuration/HousekeepLogFilesProcessConfiguration.java]: done

**Review by mick-stewart73 — APPROVED** [review]

**Pull request merged**